### PR TITLE
Manual backport for PR 11622

### PR DESCRIPTION
**Pull request by pllim**
### Description

This is a manual backport for #11622 . I am trying to see if the RTD job here points to "astropy-dev" as expected, since the doc here is not considered to be "dev".

I also noticed that `git cherry-pick` resulted in conflict (probably because LTS doc and dev doc have diverged so much), so this manual backport is unavoidable. But feel free to close without merge if the release manager finds otherwise.

**DO NOT MERGE** until after #11622 is merged.

# TODO

- [x] Remove TMP commits after RTD builds.
- [x] Is #11622 merged?
- [ ] Does release manager need this or did auto-backport work?

### CODE REVIEW

**Comment by pllim** [comment]
Azure wheel failure is unrelated.

File: docs/conf.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
 import sys
 
 import astropy
+from sphinx.errors import NoUri
```
Comment (pllim) [inline docs/conf.py:33]: Don't think this is used.

```suggestion
```

File: docs/index.rst [hunk 2]
```diff
@@ -158,8 +158,9 @@ There are some additional tools, mostly of use for maintainers, in the
 
 {%else%}
 
-To read the developer documentation, you will need to go to the
-`latest developer version of the documentation <https://docs.astropy.org/en/latest/#developer-documentation>`__.
+To read the developer documentation, you will need to go to the :ref:`latest
+developer version of the documentation
+<astropy-dev:developer-documentation>`.
```
Comment (pllim) [inline docs/index.rst:163]: This didn't render as "stable": `docs/index.rst:133: WARNING: undefined label: developer-documentation`

File: docs/install.rst [hunk 3]
```diff
@@ -148,17 +148,9 @@ this case you may consider using the ``--user`` option to install the package
...
-``astropy``, such as an affiliated package, consider installing ``astropy`` into a :ref:`virtualenv<using-virtualenv>`.
-
-{%else%}
-
 Alternatively, if you intend to do development on other software that uses
-``astropy``, such as an affiliated package, consider installing ``astropy`` into a `virtualenv <https://docs.astropy.org/en/latest/development/workflow/virtualenv_detail.html>`__.
-
-{%endif%}
+``astropy``, such as an affiliated package, consider installing ``astropy``
+into a :ref:`virtualenv <astropy-dev:virtual_envs>`.
```
Comment (pllim) [inline docs/install.rst:153]: This rendered correctly to https://docs.astropy.org/en/latest/development/workflow/virtual_pythons.html#virtual-envs

File: docs/index.rst (hunk 2 above)
Comment (embray) [reply docs/index.rst:163]: I'm not sure what you mean here.  In this case it should be using intersphinx to resolve to latest/, so long as that label actually exists in latest/

Comment (pllim) [reply docs/index.rst:163]: But it failed. Do you know why?

Comment (pllim) [reply docs/index.rst:163]: Ah, I think I figured it out. The URL points to heading title, but now we need to point to Sphinx ref tag: https://github.com/astropy/astropy/pull/11622/files#r623313262

**Review by pllim — COMMENTED** [review]
I can confirm that this renders for "stable" as expected. I will remove temporary commits and squash. Once I do that, RTD will fail again until #11466 is backported. Also, even if not, the rendering will be misleading because `dev` will resolve to `True` again until a release is cut.

tl;dr -- Just trust me.

**Comment by embray** [comment]
It looks OK now?

**Comment by pllim** [comment]
> It looks OK now?

I think so, but I'll leave it to the release manager to merge.

**Comment by astrofrog** [comment]
Oops I backported this manually/locally in 0e44d7c68ab19b6351412be4aae76299ae1acdb5 so I think we can close this?
